src/middleware.rs

```diff
@@ -13,19 +13,24 @@ use tower_sessions::Session;
 
 use openidconnect::{
     core::{CoreAuthenticationFlow, CoreErrorResponseType, CoreGenderClaim, CoreJsonWebKey},
-    AccessToken, AccessTokenHash, AuthenticationContextClass, CsrfToken, IdTokenClaims,
+    AccessToken, AccessTokenHash, Audience, AuthenticationContextClass, CsrfToken, IdTokenClaims,
     IdTokenVerifier, Nonce, OAuth2TokenResponse, PkceCodeChallenge, RefreshToken,
     RequestTokenError::ServerResponse,
-    Scope, TokenResponse,
+    Scope, TokenResponse, UserInfoClaims,
 };
 
 use crate::{
     error::MiddlewareError,
-    extractor::{OidcAccessToken, OidcClaims, OidcRpInitiatedLogout},
+    extractor::{OidcAccessToken, OidcClaims, OidcRpInitiatedLogout, OidcUserClaims},
     AdditionalClaims, AuthenticatedSession, BoxError, ClearSessionFlag, IdToken, OidcClient,
     OidcSession, SESSION_KEY,
 };
 
+#[derive(Clone, Default, Debug)]
+pub struct Config {
+    pub other_audiences: Vec<Audience>,
+}
+
```

Comment:
After reading [the OIDC spec](https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation) I think that a `trusted_audiences: Option<Vec<_>>` and `untrusted_audiences: Vec<_>` field in the `crate::builder` would be better.